packages/@ember/-internals/glimmer/lib/helpers/array.ts

```diff
@@ -43,5 +44,5 @@ import { Opaque } from '@glimmer/util';
  */
 
 export default function(_vm: VM, args: Arguments): PathReference<Opaque[]> {
-  return args.positional.capture();
+  return new EmberCapturedArrayReference(args.positional.capture());
```

Comment:
Hmm, do we need to make a whole new reference type for this? How does `{{this.foo.firstObject}}`/`{{this.foo.lastObject}}` work (unrelated to `{{array`) when `this.foo` is a "normal array"?

Why do the normal computed properties ([see here](https://github.com/emberjs/ember.js/blob/9a09374ef0f90d5239ebf2f538c902fb1fb04c12/packages/%40ember/-internals/runtime/lib/mixins/array.js#L323-L325)) for `firstObject`/`lastObject` that are added to `Array.prototype` not work automatically?

Reply:
I think the reason it does not work is: for array from component/controller it creates ObjectReference (RootReference), for array from `{{array}}` helper it is just array of references so `firstObject/lastObject` does not exist for it. So simplest way is just to create another _Reference_ to translate `firstObject/lastObject` to `[0]/[length -1]`,  if I have better solution I will make followup PR to improve it 

>  How does {{this.foo.firstObject}}/{{this.foo.lastObject}} work (unrelated to {{array) when this.foo is a "normal array"?
`{{this.foo.firstObject}}/{{this.foo.lastObject}}` should be good, since it uses `RootReference`
 there are tests covers for it too I think